projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -433,6 +466,43 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
         }
     }
 
+    /**
+     * Event handlers
+     * @hidden @internal
+     */
+    public handleOpening(event: CancelableEventArgs) {
+        const args: CancelableEventArgs = { cancel: event.cancel };
+        this.onOpening.emit(args);
+
+        event.cancel = args.cancel;
+        if (args.cancel) {
+            return;
+        }
+        this.scrollToItem(this.selectedItem);
+    }
+
+    /** @hidden @internal */
+    public handleOpened() {
+        this.updateItemFocus();
+        this.onOpened.emit();
+    }
+
+    /** @hidden @internal */
+    public handleClosing(event) {
+        const args: CancelableEventArgs = { cancel: event.cancel };
+        this.onClosing.emit(args);
+        event.cancel = args.cancel;
+        if (args.cancel) {
```

Comment:
this check here is not necessary. There is no difference if `cancel` is `true` or `false` in this case - we will go out of the method.